pkgs/development/python-modules/opensfm/default.nix

```diff
@@ -0,0 +1,106 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, opencv
+, opencv4
+, ceres-solver
+, suitesparse
+, metis
+, eigen
+, pkg-config
+, pybind11
+, numpy
+, pyyaml
+, lapack
+, gtest
+, gflags
+, glog
+, pytestCheckHook
+, networkx
+, pillow
+, exifread
+, gpxpy
+, pyproj
+, python-dateutil
+, joblib
+, repoze_lru
+, xmltodict
+, cloudpickle
+, scipy
+}:
+
+let
+  pname = "OpenSfM";
+  version = "0.5.1";
+  src = fetchFromGitHub {
+    owner = "mapillary";
+    repo = "OpenSfM";
+    rev = "v${version}";
+    sha256 = "sha256-AU3n3AVgNWeRBFt5hnuzDVU4WEWhI9mmKUuARnFBY4c=";
+    fetchSubmodules = true;
+  };
+  opensfm =
+    buildPythonPackage {
+      inherit pname version src;
+      nativeBuildInputs = [ cmake pkg-config ];
+      buildInputs = [
+        opencv
+        ceres-solver.dev
+        suitesparse
+        metis
+        eigen
+        lapack
+        gflags
+        gtest
+        glog
+      ];
+      propagatedBuildInputs = [
+        numpy
+        scipy
+        pyyaml
+        opencv4
+        networkx
+        pillow
+        exifread
+        gpxpy
+        pyproj
+        python-dateutil
+        joblib
+        repoze_lru
+        xmltodict
+        cloudpickle
+      ];
+      checkInputs = [ pytestCheckHook ];
+      dontUseCmakeBuildDir = true;
+      dontUseCmakeConfigure = true;
+      cmakeFlags = [
+        "-Bcmake_build"
+        "-Sopensfm/src"
+      ];
+      patches = [ ./fix-cmake.patch ];
```

Comment:
Can we fetch this from upstream? It's there a chance we could upstream this?

Reply:
This cannot be fetched from upstream. The upstream is using custom `Find*.cmake` modules and distributes an intree copy of `gtest` which breaks automatic cmake package discovery. I'm deleting these `Find*` files and the gtest copy, adjusting the root `CMakeLists.txt` accordingly

We cannot upstream this right away, because it's a lot of changes I might struggle to defend, but I'll definitely try to get in touch with them

Reply:
@SuperSandro2000 may I mark these two comments as resolved?